generators/client/generator.mjs

```diff
@@ -312,4 +313,29 @@ export default class JHipsterClientGenerator extends BaseApplicationGenerator {
   get [BaseApplicationGenerator.END]() {
     return this.asEndTaskGroup(this.delegateTasksToBlueprint(() => this.end));
   }
+
+  /**
+   * @private
+   * Add a new icon to icon imports.
+   *
+   * @param {string} iconName - The name of the Font Awesome icon.
+   * @param {string} clientFramework - The name of the client framework
+   */
+  addIcon(iconName, clientFramework) {
+    addIconInImport(this, iconName, clientFramework);
+  }
+
+  /**
+   * @private
+   * Add a new menu element, at the root of the menu.
+   *
+   * @param {string} routerName - The name of the Angular router that is added to the menu.
+   * @param {string} iconName - The name of the Font Awesome icon that will be displayed.
+   * @param {boolean} enableTranslation - If translations are enabled or not
+   * @param {string} clientFramework - The name of the client framework
+   * @param {string} translationKeyMenu - i18n key for entry in the menu
+   */
+  addElementToMenu(routerName, iconName, enableTranslation, clientFramework, translationKeyMenu = _.camelCase(routerName)) {
+    addMenuEntry(this, routerName, iconName, enableTranslation, clientFramework, translationKeyMenu);
+  }
```

Comment:
Needles are client framework specific, and should change the parameters to 1 typed implementation

Reply:
Ok but for now I just moved the code as-is.